### ISSUE TITLE
Running nixos tests in nixpkgs fails on Darwin, even with a remote builder set up.

### ISSUE THREAD

**Issue by khaled** [NixOS/nixpkgs#304580]
### Describe the bug
Running nixos tests fails in on darwin with a "package xyz is not available on the requested hostPlatform" error.

### Steps To Reproduce
Set up a [darwin linux builder](https://nixos.org/manual/nixpkgs/stable/#sec-darwin-builder).

Then run one of the nixos tests in nixpkgs.  For example:

```
nix build -L  nixpkgs#nixosTests.k3s.single-node.k3s_1_29
```

### Expected behavior
The test runs successfully.

### Additional context
I am generally able to build linux artifacts and run nixos VMs.  And I'm able to run the [example here](https://discourse.nixos.org/t/macos-support-for-running-nixos-tests/40801).  But I cannot successfully run any of the tests in nixpkgs.

### Notify maintainers
@Gabriella439 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/rbkcv82pwkdfksh3qq3v5w0ha1nlzm84-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Gabriella439** [comment]
Yeah, so this *might* be an issue with the fact that these tests use `nixos/tests/make-test-python.nix` rather than `pkgs.nixosTest`.  I only verified that the `pkgs.nixosTest` path works correctly, but the `make-test-python.nix` entrypoint might need additional work.

**Comment by khaled** [comment]
Thanks @Gabriella439!

@superherointj I'm not sure it's right to tag this as k3s specific (topic label + issue title), because apparently there are [ton of tests](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20make-test-python.nix&type=code) using `make-test-python.nix`.

**Comment by superherointj** [comment]
> Thanks @Gabriella439!
> 
> @superherointj I'm not sure it's right to tag this as k3s specific (topic label + issue title), because apparently there are [ton of tests](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20make-test-python.nix&type=code) using `make-test-python.nix`.

Sorry. I must have made some confusion. Wrong tab, idk.
